hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveStatelessSessionImpl.java

```diff
@@ -0,0 +1,418 @@
+                    .getSQLParameterMetadata(sqlString, false);
+            ReactiveNativeQueryImpl<T> query =
+                    new ReactiveNativeQueryImpl<>(sqlString, false, this, params );
+            query.setComment( "dynamic native SQL query" );
+            applyQuerySettingsAndHints( query );
+            return query;
+        }
+        catch ( RuntimeException he ) {
+            throw getExceptionConverter().convert( he );
+        }
+    }
+
+    @Override
+    public <T> ReactiveNativeQuery<T> createReactiveNativeQuery(String sqlString, Class<T> resultClass) {
+        try {
+            ReactiveNativeQuery<T> query = createReactiveNativeQuery( sqlString );
+            handleNativeQueryResult( query, resultClass );
+            return query;
+        }
+        catch ( RuntimeException he ) {
+            throw getExceptionConverter().convert( he );
+        }
+    }
+
+    @Override
+    public <T> ReactiveNativeQuery<T> createReactiveNativeQuery(String sqlString, String resultSetMapping) {
+        try {
+            ReactiveNativeQuery<T> query = createReactiveNativeQuery( sqlString );
+            query.setResultSetMapping( resultSetMapping );
+            return query;
+        }
+        catch ( RuntimeException he ) {
+            throw getExceptionConverter().convert( he );
+        }
+    }
+
+    @Override
+    public <T> CompletionStage<List<T>> reactiveList(String query, QueryParameters parameters) {
+        checkOpen();
+        parameters.validateParameters();
+
+        HQLQueryPlan plan = parameters.getQueryPlan();
+        if ( plan == null ) {
+            plan = getQueryPlan( query, false );
+        }
+        ReactiveHQLQueryPlan reactivePlan = (ReactiveHQLQueryPlan) plan;
+
+        return reactivePlan.performReactiveList(parameters, this )
+                .whenComplete( (list, x) -> {
+                    getPersistenceContext().clear();
+                    afterOperation( x == null );
+                } )
+                //TODO: this typecast is rubbish
+                .thenApply( list -> (List<T>) list );
+    }
+
+    @Override
+    public <T> CompletionStage<List<T>> reactiveList(NativeSQLQuerySpecification spec, QueryParameters parameters) {
+        checkOpen();
+
+        ReactiveCustomLoader loader = new ReactiveCustomLoader(
+                getNativeQueryPlan( spec ).getCustomQuery(),
+                getFactory()
+        );
+
+        return loader.reactiveList(this, parameters)
+                .whenComplete((r, x) -> {
+                    getPersistenceContext().clear();
+                    afterOperation(x == null);
+                })
+                //TODO: this typecast is rubbish
+                .thenApply( list -> (List<T>) list );
+    }
+
+    @Override
+    public CompletionStage<Integer> executeReactiveUpdate(String expandedQuery, QueryParameters parameters) {
+        throw new UnsupportedOperationException();
```

Comment:
Are these going to be implemented later, or an implicit limitation of this session kind?

Reply:
Pretty sure it's I just `cos I didn't implement them yet.

(This is for native and HQL `update` and `delete` queries, FTR.)

Reply:
I've implemented these methods.